assets/src/edit-story/components/canvas/pageAttachment/index.js

```diff
@@ -0,0 +1,177 @@
+ */
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useCanvas from '../useCanvas';
+import Popup from '../../popup';
+import { useTransform } from '../../transform';
+
+const Wrapper = styled.div`
+  position: absolute;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: column;
+  bottom: 0;
+  height: 20%;
+  width: 100%;
+  color: ${({ theme }) => theme.colors.fg.v1};
+  z-index: 3;
+`;
+
+const Guideline = styled.div`
+  mix-blend-mode: difference;
+  position: absolute;
+  height: 1px;
+  bottom: 20%;
+  width: 100%;
+  background-image: ${({ theme }) =>
+    `linear-gradient(to right, ${theme.colors.fg.v0} 50%, ${theme.colors.fg.v1} 0%)`};
+  background-position: top;
+  background-size: 16px 0.5px;
+  background-repeat: repeat-x;
+  z-index: 3;
+`;
+
+// The CSS here is based on how it's displayed in the front-end, including static
+// font-size, line-height, etc. independent of the viewport size -- it's not responsive.
+const Icon = styled.div`
+  position: relative;
+  display: block;
+  height: 28px;
+  width: 32px;
+`;
+
+const arrowBarCss = css`
+  position: absolute;
+  display: block;
+  height: 3px;
+  width: 12px;
+  border-radius: 3px;
+  top: 14px;
+  background: ${({ theme }) => theme.colors.fg.v1};
+  box-shadow: 0px 0px 6px rgba(0, 0, 0, 0.36);
+`;
+
+const LeftBar = styled.div`
+  ${arrowBarCss}
+  left: 6px;
+  transform: rotate(-30deg);
+`;
+
+const RightBar = styled.div`
+  ${arrowBarCss}
+  right: 6px;
+  transform: rotate(30deg);
+`;
+
+const TextWrapper = styled.span`
+  font-family: Roboto;
+  font-size: 16px;
+  line-height: 16px;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  text-shadow: 0px 0px 6px rgba(0, 0, 0, 0.36);
+  white-space: nowrap;
+  max-width: 90%;
+  position: relative;
+  padding: 0 32px;
+  height: 16px;
+  letter-spacing: 0.3px;
+`;
+
+const Tooltip = styled.div`
+  background-color: ${({ theme }) => theme.colors.bg.v0};
+  color: ${({ theme }) => theme.colors.fg.v1};
+  width: 200px;
+  padding: 8px;
+  font-size: 14px;
+  border-radius: 4px;
+  text-align: center;
+`;
+
+const spacing = { x: 8 };
+
+function PageAttachment({ pageAttachment = {} }) {
+  const {
+    hasLinkInAttachmentArea,
+    pageAttachmentContainer,
+    setPageAttachmentContainer,
+  } = useCanvas((state) => ({
+    hasLinkInAttachmentArea: state.state.hasLinkInAttachmentArea,
+    pageAttachmentContainer: state.state.pageAttachmentContainer,
+    setPageAttachmentContainer: state.actions.setPageAttachmentContainer,
+  }));
+  const {
+    state: { isAnythingTransforming },
+  } = useTransform();
+  const { ctaText = __('Learn more', 'web-stories'), url } = pageAttachment;
+  return (
+    <>
+      {hasLinkInAttachmentArea && <Guideline />}
+      <Wrapper role="presentation" ref={setPageAttachmentContainer}>
```

Comment:
Clever - as the `ref` will be invoked with `null` when the particular element unmounts.